src/Cartography/Cartography.DynamicMap/Platforms/Android/Clustering/MapControlBase.Cluster.Android.cs

```diff
@@ -0,0 +1,129 @@
+﻿#if __ANDROID__
+using Android.Gms.Maps;
+using Android.Gms.Maps.Utils.Clustering;
+using Chinook.DynamicMvvm;
+using GeolocatorService;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reactive.Linq;
+using System.Text;
+using Uno.Extensions;
+using Uno.Logging;
+using Windows.UI.Xaml;
+
+namespace Cartography.DynamicMap;
+
+public partial class MapControlBase
+{
+    private CustomClusterManager _clusterManager;
+    private MapClusterItems _clusterPins;
+
+    private void OnMapClusterReady()
+    {
+        // Zoomlevel does not need to be manage here. ClusterManager will handle how and when a cluster is shown. 
+        // A cluster can show a single pin. The diffrent styles will be manage in ClusterManager.
+
+        _clusterPins = new MapClusterItems(new MapClusterItem[0]);
+        _clusterManager = new CustomClusterManager(Context, _map);
+        SetupCustomClusterManager(_clusterManager);
+
+        UpdateMapPushpinOnCameraIdle();
+
+        _isReady = true;
+
+        UpdateAutolocateButtonVisibility(AutolocateButtonVisibility);
+        UpdateCompassButtonVisibility(CompassButtonVisibility);
+        UpdateIsRotateGestureEnabled(IsRotateGestureEnabled);
+        UpdateMapStyleJson(MapStyleJson);
+
+        UpdateIcon(PushpinIcon);
+        UpdateSelectedIcon(SelectedPushpinIcon);
+
+        TryStart();
+    }
+
+    private void SetupCustomClusterManager(CustomClusterManager cluster)
+    {
+        _map.SetOnMarkerClickListener(cluster);
+        _map.SetOnInfoWindowClickListener(cluster);
+        cluster.SetOnClusterClickListener(cluster);
+        cluster.SetOnClusterInfoWindowClickListener(cluster);
+        cluster.SetOnClusterItemClickListener(new ClusterItemClick(this));
+        cluster.SetOnClusterItemInfoWindowClickListener(cluster);
+    }
+
+    private void UpdateAndroidClusteringPushpins(IGeoLocated[] items, IGeoLocated[] selectedItems)
+    {
+        if (_clusterPins != null)
+        {
+            var pinsToAdd = ClusterItemsToAdd(items, false);
+            var pinsToRemove = ClusterItemsToRemove(items, false);
+            _clusterPins.UpdateItems(pinsToAdd, pinsToRemove);
+
+            var selectedPinsToAdd = ClusterItemsToAdd(selectedItems, true);
+            var selectedPinsToRemove = ClusterItemsToRemove(selectedItems, true);
+            _clusterPins.UpdateItems(selectedPinsToAdd, selectedPinsToRemove);
+
+            _clusterManager.RemoveItems(pinsToRemove);
+            _clusterManager.RemoveItems(selectedPinsToRemove);
+            _clusterManager.AddItems(pinsToAdd);
+            _clusterManager.AddItems(selectedPinsToAdd);
+            _clusterManager.Cluster();
+        }
+    }
+
+    private MapClusterItem[] ClusterItemsToAdd(IGeoLocated[] items, bool isSelected)
```

Comment:
This method (and the one below) should be prefixed by "Get".
Any method should start with a verb.